### PR TITLE
ci: build fewer LLVM components

### PR DESCRIPTION
**Pull request by abrown**
This is a recommendation from #304 to build less of LLVM in CI. Perhaps it can speed up build times.

### CODE REVIEW

**Comment by abrown** [comment]
cc: @turbolent

File: Makefile [hunk 1]
```diff
@@ -64,6 +64,10 @@ build/llvm.BUILT:
 		-DCMAKE_OSX_ARCHITECTURES="arm64;x86_64" \
 		-DCMAKE_OSX_DEPLOYMENT_TARGET=11.0 \
 		-DCMAKE_INSTALL_PREFIX=$(PREFIX) \
+		-DLLVM_INCLUDE_TESTS=OFF \
+		-DLLVM_INCLUDE_UTILS=OFF \
```
[the remark this answers is not in the document]
Comment (abrown) [reply Makefile:69]: Not from what I see. Searching through `llvm-project` I see it conditioning things in `utils` like:

```
FileCheck
PerfectShuffle
count
not
UnicodeData
yaml-bench
split-file
LLVMVisualizers
llvm-lit
llvm-locstats
llvm-jitlink-executor
```

I might have missed some, but they don't like the tools one would expect to see in wasi-sdk:

```
$ ls /opt/wasi-sdk/wasi-sdk-20.0/bin
ar        clang-apply-replacements  git-clang-format  llvm-ar         llvm-objcopy  llvm-strip  run-clang-tidy
c++filt   clang-cl                  ld64.lld          llvm-cxxfilt    llvm-objdump  nm          size
clang     clang-cpp                 ld.lld            llvm-dwarfdump  llvm-ranlib   objcopy     strings
clang++   clang-format              lld               llvm-mc         llvm-size     objdump     strip
clang-16  clang-tidy                lld-link          llvm-nm         llvm-strings  ranlib      wasm-ld
```

I'm going to rebase this, see if the CI run is any shorter, and merge if it is.

[the remark this answers is not in the document]
Comment (sbc100) [reply Makefile:69]: sgtm

**Comment by abrown** [comment]
Just to make sure this change didn't remove anything, I quickly checked that we're at least packaging up the same number of files:

```console
$ ci/download-workflow-artifacts.sh 495322858 $TOKEN
$ tar tvf /tmp/wasi-sdk-artifacts.JJo0nuS/release/wasi-sdk-20.8g74cb50e13861-linux.tar.gz | wc -l
1426
$ find /opt/wasi-sdk/wasi-sdk-20.0 | wc -l
1426
```

**Pull request merged**